### PR TITLE
completed task 6 awaiting review - implemented jest sorted for task 4…

### PR DESCRIPTION
**Pull request by lphamilton87**
… and 6

core task 6
branch name: nc-news-task-06

For task 6 I used jest sorted for testing, I also went back and implemented it for my tests for task 4.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -101,3 +110,47 @@ describe("GET/api/articles/:id", () => {
...
   });
 });
+
+describe("GET/api/articles/:article_id/comments", () => {
+  test("returns 200: an array of comments for the given article_id", () => {
+    return request(app)
+      .get("/api/articles/3/comments")
+      .expect(200)
+      .then((result) => {
+        console.log(result.body);
```
Comment (HeatherEver) [inline __tests__/app.test.js:120]: remove console logs :)

File: __tests__/app.test.js [hunk 2]
```diff
@@ -101,3 +110,47 @@ describe("GET/api/articles/:id", () => {
...
+        result.body.comments.forEach((comment) => {
+          expect(comment).toHaveProperty("comment_id", expect.any(Number));
+          expect(comment).toHaveProperty("votes", expect.any(Number));
+          expect(comment).toHaveProperty("created_at", expect.any(String));
+          expect(comment).toHaveProperty("author", expect.any(String));
+          expect(comment).toHaveProperty("body", expect.any(String));
+          expect(comment).toHaveProperty("article_id", expect.any(Number));
+        });
+      });
```
Comment (HeatherEver) [inline __tests__/app.test.js:129]: if using a forEach could you also check the length of the comments coming back is what we expect as we could get a false positive with just a forEach

File: __tests__/app.test.js [hunk 3]
```diff
@@ -101,3 +110,47 @@ describe("GET/api/articles/:id", () => {
...
+  });
+  test("returns in order with the most recent comments first", () => {
+    return request(app)
+      .get("/api/articles/1/comments")
+      .expect(200)
+      .then((response) => {
+        const result = response.body.comments;
+        expect(result).toBeSorted({ descending: "true", key: "created_at" });
+      });
+  });
```
Comment (HeatherEver) [inline __tests__/app.test.js:155]: nice testing here! We should also have a test for an article which exists but has no comments - status 200

File: db/model.js [hunk 4]
```diff
@@ -31,3 +31,14 @@ exports.fetchArticleId = (article_id) => {
...
 };
+
+exports.fetchComments = (article_id) => {
+  const query = `SELECT * FROM comments
+  WHERE comments.article_id=$1
+  ORDER BY created_at DESC`;
+  return db.query(query, [article_id]).then((comments) => {
+    if (comments.rows.length === 0) {
+      return Promise.reject({ status: 404, msg: "Not found" });
+    } else return comments.rows;
```
Comment (HeatherEver) [inline db/model.js:42]: nice work but this might need a little refactor to get the test i suggested to pass :)

**Review by HeatherEver — APPROVED** [review]
Nice work! there's just one test suggestion for you, in which case think about how if no comments come back, check that the article exists and if it does it just means it has no comments so our request will have worked the comments will just be an empty array :)

File: __tests__/app.test.js (hunk 2 above)
Comment (itskatherine) [reply __tests__/app.test.js:129]: Nice you've done this 👍

File: __tests__/app.test.js [hunk 5]
```diff
@@ -101,3 +110,105 @@ describe("GET/api/articles/:id", () => {
...
+          expect(comment).toHaveProperty("article_id", expect.any(Number));
+        });
+      });
+  });
+  test("returns an empty array when there are no comments on the article", () => {
+    return request(app)
+      .get("/api/articles/2/comments")
+      .expect(200)
+      .then((response) => {
+        result = response.body.comments;
```
Comment (itskatherine) [inline __tests__/app.test.js:136]: Don't forget `const`

File: db/controller.js [hunk 6]
```diff
@@ -24,3 +30,23 @@ exports.getArticleById = (request, response, next) => {
     })
     .catch(next);
 };
+
+exports.getComments = (request, response, next) => {
+  const articleId = request.params.article_id;
+  Promise.all([fetchComments(articleId), fetchArticleId(articleId)])
+    .then(([result]) => {
+      console.log(result);
```
Comment (itskatherine) [inline db/controller.js:38]: Nice solution to the comments problem, but don't forget to remove your console logs for PRs 👀

File: __tests__/app.test.js [hunk 7]
```diff
@@ -101,3 +110,105 @@ describe("GET/api/articles/:id", () => {
...
+      username: "rogersop",
+      body: "This is a test comment",
+    };
+    return request(app)
+      .post("/api/articles/2/comments")
+      .send(testComment)
+      .expect(201)
+      .then((response) => {
+        const result = response.body;
+        console.log(result);
```
Comment (itskatherine) [inline __tests__/app.test.js:179]: Remove console.log plsss

File: db/controller.js [hunk 8]
```diff
@@ -24,3 +30,23 @@ exports.getArticleById = (request, response, next) => {
...
+      response.status(200).send({ comments: result });
+    })
+    .catch(next);
+};
+
+exports.postNewComments = (request, response, next) => {
+  const { article_id } = request.params;
+  const { username, body } = request.body;
+  insertComments(article_id, { username, body })
+    .then((newComms) => {
```
Comment (itskatherine) [inline db/controller.js:48]: This is nit-picky of me but this is sending back a single comment so should be `newComm` not `newComms`

**Review by itskatherine — COMMENTED** [review]
Added a few comments, I see that the tickets have kind of overlapped each other, but I'll check your PR for 7 now. This branch is good to merge once you've addressed the comments tho 👍

**Pull request merged**